File: easy/ts/1431.ts
// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/?envType=study-plan-v2&envId=leetcode-75
// Time: O(n)
// Space: O(1)
function kidsWithCandies(candies: number[], extraCandies: number): boolean[] {
  let largest = 0;
  let answer: boolean[] = [];
  for (let i = 0; i < candies.length; i++) {
    if (candies[i] > largest) {
      largest = candies[i];
    }
  }

  for (let i = 0; i < candies.length; i++) {
    if (candies[i] + extraCandies >= largest) {
      answer.push(true);
    } else {
      answer.push(false);
    }
  }
  return answer;
}

File: easy/python/1431.py
# https://leetcode.com/problems/kids-with-the-greatest-number-of-candies/?envType=study-plan-v2&envId=leetcode-75
# Time: O(n)
# Space: O(1)
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        largest = 0
        answer = []

        for ele in candies:
            if ele > largest:
                largest = ele
        for ele in candies:
            if ele + extraCandies >= largest:
                answer.append(True)
            else:
                answer.append(False)
        return answer
